api/provision/consume-provisioning-job.ts

```diff
@@ -45,25 +18,9 @@ export async function baseHandler(_event: APIGatewayProxyEvent): Promise<APIGate
       cspInvocation,
       cspResponse: cspResponse.Payload,
     };
-
-    // messages being returned to client
-    messages.push(provisioningRequest);
-
-    // deleting message after processing
-    const deleteInput = { QueueUrl: PROVISIONING_QUEUE_URL, ReceiptHandle: record.ReceiptHandle };
-    const deleteMessageCommand = new DeleteMessageCommand(deleteInput);
-    const deletedMessageResponse = await sqsClient.send(deleteMessageCommand);
-    logger.info("Message Deleted", { deletedMessageResponse } as any);
   }
-
-  logger.info(`Number of Messages processed ${messages.length}`);
-  return new ApiSuccessResponse(messages, SuccessStatusCode.OK);
 }
 
-export const handler = middy(baseHandler)
-  .use(injectLambdaContext(logger))
-  .use(inputOutputLogger({ logger: (message) => logger.info("Event/Result", message) }))
-  .use(errorLogger({ logger: (err) => logger.error("An error occurred during the request", err as Error) }))
-  .use(IpCheckerMiddleware())
-  .use(errorHandlingMiddleware())
-  .use(JSONErrorHandlerMiddleware());
+const handlerClass = new ConsumeProvisioningJob(PROVISIONING_QUEUE_URL);
+
+export const handler = handlerClass.createHandler();
```

Comment:
I love this

Reply:
Thanks. It took a little bit of futzing around to get it to work (e.g. since it's a closure, having it rely on a member variable like `queueUrl` wasn't going to work). Was a fun problem to solve though.